homeassistant/helpers/template.py

```diff
@@ -1043,6 +1043,20 @@ def _collect_state(self) -> None:
         if self._collect and (render_info := _render_info.get()):
             render_info.entities.add(self._entity_id)  # type: ignore[attr-defined]
 
+    def _get_translated_unit_of_measurement(self, language: str) -> str | None:
+        entry = entity_registry.async_get(self._hass).async_get(self.entity_id)
+        if entry is None:
+            return None
+        platform = entry.platform
+        translation_key = entry.translation_key
+        if platform is None or translation_key is None:
+            return None
+        localize_key = f"component.{platform}.entity.{self._state.domain}.{translation_key}.unit_of_measurement"
+        translations = async_get_cached_translations(self._hass, language, "entity")
+        if localize_key not in translations:
+            return None
+        return translations[localize_key]
```

Comment:
I'd suggest the translation logic is kept in the translation helper instead of here, similar to how we call `async_translate_state` instead of implement it here